IGCexpansion/CodonGeneconv.py

```diff
@@ -653,11 +684,32 @@ def loglikelihood_and_gradient(self, display = False):
                 if i in self.Force.keys():  # check here
                     other_derivs.append(0.0)
                     continue
+            # x_plus_delta = np.array(self.x)
+            # x_plus_delta[i] += delta
+            # self.update_by_x(x_plus_delta)
+            # ll_delta, _ = fn(store = True, edge_derivative = False)
+            # d_estimate = (ll_delta - ll) / delta
+
+            # finite difference central
+
+            delta = deepcopy(max(1, abs(self.x[i])) * 0.000001)

```

Comment:
Do you need `deepcopy` for a scaler?

Reply:
I think you are right. I was worried about self.x might change, but I realize maybe it is not necessary.